include/openthread/border_agent.h

```diff
@@ -123,6 +123,36 @@ bool otBorderAgentIsActive(otInstance *aInstance);
  */
 uint16_t otBorderAgentGetUdpPort(otInstance *aInstance);
 
+/**
+ * This callback informs the application the changes of the state of the MeshCoP service.
+ *
+ * In specific, the 'state' includes the MeshCoP TXT data originated from the Thread network and whether the
+ * Border Agent is Active (which can be obtained by `otBorderAgentIsActive`).
+ *
+ * @param[in] aTxtData  A pointer to the encoded MeshCoP TXT data originated from the Thread network.
+ * @param[in] aLength   The length of the encoded TXT data.
+ * @param[in] aContext  A pointer to application-specific context.
+ */
+typedef void (*otBorderAgentMeshCoPServiceChangedCallback)(const uint8_t *aTxtData, uint16_t aLength, void *aContext);
```

Comment:
context is usually either the first or last parameter in a callback. Since some callbacks may take variadic arguments, perhaps we should make a convention to put it as the first parameter. This may help for building some template functions in future. Thoughts? @abtink

Reply:
Unfortunately both ways (context first or last) are already used by public OT APIs. 
[`Callback` class](https://github.com/openthread/openthread/blob/a2d65a1ad78ca8b1e6dc12cb0241812190032d70/src/core/common/callback.hpp) already handles both ways.